frontend/packages/core-api-client/src/client/Record.ts

```diff
@@ -11,16 +11,58 @@ import {
   RecordLookup,
   FormDefinition,
   FieldValue,
+  RecordClientDefinition,
+  FormWithRecord,
 } from '../types';
 
 import { BaseRESTClient } from './BaseRESTClient';
 import { FormClient } from './Form';
 
-export class RecordClient {
+export class RecordClient implements RecordClientDefinition {
   private restClient: BaseRESTClient;
 
   private formClient: FormClient;
 
+  static buildDefaultRecord = (form: FormDefinition): Record => ({
```

Comment:
what's your reason for putting this here? to me, a client is purely for communication with the api.

Reply:
I think of the client more for interacting with the "model" rather than just the api. Things that are generic enough for the `Record` model I would add here. This seems like a useful method to have across applications.

For example, I wouldn't add the code to transform records to react-hook-forms as that is specific to the client implementation.